server/src/main/java/com/thoughtworks/go/server/domain/PluginSettings.java

```diff
@@ -117,14 +96,18 @@ public boolean hasErrors() {
     public Map<String, String> getSettingsAsKeyValuePair() {
         Map<String, String> settingsAsKeyValuePair = new HashMap<>();
         for (ConfigurationProperty configurationProperty : settingsMap) {
-            settingsAsKeyValuePair.put(configurationProperty.getConfigKeyName(), configurationProperty.getConfigValue());
+            settingsAsKeyValuePair.put(configurationProperty.getConfigKeyName(), configurationProperty.getValue());
         }
         return settingsAsKeyValuePair;
     }
 
-    public void populateSettingsMap(Plugin plugin) {
+    public void populateSettingsMap(Plugin plugin, PluginInfo pluginInfo) {
+        final ConfigurationPropertyBuilder builder = new ConfigurationPropertyBuilder();
         for (String settingsKey : plugin.getAllConfigurationKeys()) {
-            settingsMap.add(new ConfigurationProperty(new ConfigurationKey(settingsKey), new ConfigurationValue(plugin.getConfigurationValue(settingsKey))));
+            final PluginConfiguration configuration = pluginInfo.getPluginSettings().getConfiguration(settingsKey);
+            if (configuration != null) {
```

Comment:
Same here, lets not ignore unknown configurations

Reply:
done